### PR TITLE
Switch from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by nmaludy**
Travis is slow... GitHub actions is fast... need i say more :)

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.puppet }}-${{ matrix.ruby }}-pip-unit-
+      - name: Python prep
+        run: |
+          pip install virtualenv
+      - name: Test
+        run: 'make ${{ matrix.make_target }}'
+
+
+# TODO create a job for doing integration tests
```
Comment (armab) [inline .github/workflows/build.yaml:110]: Tried to find this in CI checks, but figured the WIP.
That would be the most interesting part to compare and see how it works! :+1:

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+          bundle install --jobs $(nproc) --retry 3 
+      - name: Test
+        run: |
+          bundle exec kitchen test --debug ${{ matrix.name }}
+
+  deploy:
+    name: 'deploy to forge'
+    needs:
+      - unit
+      - integration
```
Comment (armab) [inline .github/workflows/build.yaml:172]: :+1: 

Does it make sense to do something similar for the Integration tests?
Eg. start them only when lint checks and unit tests succeeded?

That can potentially save some computing resources if there was an issue with lint/unit.
Just trying to be a good OSS citizen :)

Comment (nmaludy) [reply .github/workflows/build.yaml]: That's fair, fixed!

I also noticed that i was not including the `unit-python` in this dependency graph.!

**Comment by nmaludy** [comment]
@armab i fixed the cache issue i was having for the integration tests, this is now good to go!

**Review by armab — APPROVED** [review]
Good stuff! :+1: 

Do you plan to remove the Travis config as well?

**Comment by nmaludy** [comment]
@armab going to keep it around for a short period of time in case we want to switch back, i will remove in the future though when we are confident we want to stick with it... for now it's as easy as renaming that file to turn travis builds back on

**Pull request merged**